SpatialGDK/Source/SpatialGDKEditor/Public/SpatialOSLoadBalancingStrategies.h

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "UObject/Object.h"
+#include "Serialization/JsonWriter.h"
+
+#include "SpatialOSLoadBalancingStrategies.generated.h"
+
+UCLASS(Abstract)
+class SPATIALGDKEDITOR_API USpatialOSAbstractLoadBalancingStrategy : public UObject
+{
+	GENERATED_BODY()
+
+public:
+	virtual bool WriteToConfiguration(TSharedRef<TJsonWriter<>> Writer) const
+	{
+		check(0); return false;
+	}
+
+	virtual int32 GetNumberOfWorkersForPIE() const
+	{
+		check(0); return false;
+	}
```

Comment:
should these be pure virtual instead?

Reply:
I tried, but Unreal insists on trying to instantiate them. Maybe I do not know the good macro to allow an abstract UObject definition ?

Reply:
Did you try the PURE_VIRTUAL macro?

Reply:
AFAIK the PURE_VIRTUAL macro would expand to something similar, or to = 0 depending on build options.

Reply:
the PURE_VIRTUAL macro is better for expressing intent.

Reply:
Yep, but I'm afraid it might break if we get a build that makes the macro evaluates to = 0